lib/FrontendTool/FrontendTool.cpp

```diff
@@ -1623,20 +1623,37 @@ int swift::performFrontend(ArrayRef<const char *> Args,
   // arguments are generated by the driver, not directly by a user. The driver
   // is responsible for emitting diagnostics for its own errors. See SR-2683
   // for details.
-  //
-  // FIXME: Do we need one SerializedConsumer per primary? Owned by the
-  // SourceFile?
-  std::unique_ptr<DiagnosticConsumer> SerializedConsumer;
+  std::unique_ptr<DiagnosticConsumer> SerializedConsumerMultiplexer;
   {
-    const std::string &SerializedDiagnosticsPath =
-        Invocation.getSerializedDiagnosticsPathForAtMostOnePrimary();
-    if (!SerializedDiagnosticsPath.empty()) {
-      SerializedConsumer.reset(
+    SmallVector<FileSpecificDiagnosticConsumer::ConsumerPair, 4>
+      SerializedConsumers;
+    Invocation.getFrontendOptions().InputsAndOutputs
```

Comment:
Is it good practice to add a `(void)` in front of this line to show that the code explicitly throws away the result?

Reply:
Hm. Maybe? I feel like this is one of those cases where the result was always incidental to begin with, like the result of `Dictionary.remove` in Swift. Some situations care about it, some don't, but ignoring it is intended much more often than it's a mistake.